lib/sinatra/tracer.rb

```diff
@@ -1,7 +1,39 @@
-require "sinatra/tracer/version"
+require 'rack/tracer'
+require 'sinatra/base'
+require 'sinatra/tracer/version'
+require 'opentracing'
 
 module Sinatra
   module Tracer
-    # Your code goes here...
+
+    class << self
+
+      def registered(app)
+        # enable the Rack Tracer middleware
+        app.use Rack::Tracer
+
+        patch_render
+      end
+
+      def patch_render
+        ::Sinatra::Base.module_eval do
+          alias_method :render_original, :render
+
+          def render(engine, data, options = {}, locals = {}, &block)
+            result = ""
+
+            OpenTracing.global_tracer.start_active_span("Rendering View") do |scope|
+              result = render_original(engine, data, options, locals, &block)
+              scope.span.set_tag("sinatra.template", data)
```

Comment:
What's `data` here?

Reply:
`data` is the name of the template or the template code itself if the caller is rendering a string. I can add an item in the readme to document this tag